src/Storages/MergeTree/MergeTreeBaseSelectProcessor.cpp

```diff
@@ -628,12 +628,12 @@ Block MergeTreeBaseSelectProcessor::transformHeader(
         else
         {
             WhichDataType which(removeNullable(recursiveRemoveLowCardinality(prewhere_column.type)));
-            if (which.isInt() || which.isUInt())
+            if (which.isNativeInt() || which.isNativeUInt())
                 prewhere_column.column = prewhere_column.type->createColumnConst(block.rows(), 1u)->convertToFullColumnIfConst();
```

Comment:
This solution avoids the crash, but what we can also do is add special cases for non-native ints but it will fail regardless at a later point so I just left it like this.